File: cpp/1281.cpp
#include <stdio.h>
#include <iostream>
#include <string>
#include <map>

using namespace std;

int main()
{   
    int n;
    cin >> n;

    for (n; n > 0; n--) {
        map<string, float> items;
        int m, p, amount;
        string item;
        float price, sum = 0;
        cin >> m;

        for (m; m > 0; m--) {
            cin >> item >> price;
            items[item] = price;
        }

        cin >> p;

        for (p; p > 0; p--) {
            cin >> item >> amount;
            sum += (items.find(item)->second) * amount;
        }

        printf("R$ %.2f\n", sum);
    }

    return 0;
}

File: py/1281.py
n = int(input())

while n > 0:
    items = dict()
    values = []

    m = int(input())
    
    while m > 0:
        item, price = input().split()
        price = float(price)

        items[item] = price
        m -= 1

    p =int(input())

    while p > 0:
        item, quantity = input().split()
        quantity = int(quantity)

        values.append(items[item] * quantity)

        p -= 1

    print(f'R$ {(sum(values)):.2f}')

    n -= 1